drizzle-orm/src/sql/expressions/conditional.ts

```diff
@@ -0,0 +1,38 @@
+import type { Column, GetColumnData } from "~/column.ts";
+
+import { type SQL, sql, type SQLWrapper } from "../sql.ts";
+
+type ExpressionType<T> = T extends Column ? GetColumnData<T, "raw"> : T extends SQL<infer I> ? I : unknown;
+
+export const caseWhen = (condition: SQL | undefined) => {
+  const sqlChunks: SQL[] = [];
+  sqlChunks.push(sql`CASE `);
+
+  const when = <AccType = null>(condition: SQL | undefined) => {
+    sqlChunks.push(sql`WHEN ${condition} `);
+
+    const end = <FinalType = null>() => {
+      sqlChunks.push(sql`END`);
+      return sql.join(sqlChunks) as SQL<FinalType>;
+    };
+
+    return {
+      then: <T extends SQL | SQLWrapper | Column>(value: T) => {
```

Comment:
overloading `.then()` seems a bit fishy to me

Reply:
It is, any other idea?

Reply:
why not when(condition, value)... something more similar to [this suggestion](https://github.com/drizzle-team/drizzle-orm/issues/1065#issuecomment-1863219952)?